packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -0,0 +1,275 @@
+import React, { useEffect, useState, useRef, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useId } from '../helpers/useId';
+import DropdownIcon from '@strapi/icons/FilterDropdownIcon';
+import {
+  getActionFromKey,
+  getUpdatedIndex,
+  isScrollable,
+  maintainScrollVisibility,
```

Comment:
we use a library to handle these specific scenario, see https://github.com/strapi/parts/blob/develop/packages/strapi-parts/package.json#L51